cvmfs/cvmfs.cc

```diff
@@ -721,20 +722,19 @@ static void cvmfs_releasedir(fuse_req_t req, fuse_ino_t ino,
 
   int reply = 0;
 
-  pthread_mutex_lock(&lock_directory_handles_);
-  DirectoryHandles::iterator iter_handle =
-    directory_handles_->find(fi->fh);
-  if (iter_handle != directory_handles_->end()) {
-    if (iter_handle->second.capacity == 0)
-      smunmap(iter_handle->second.buffer);
-    else
-      free(iter_handle->second.buffer);
-    directory_handles_->erase(iter_handle);
-    pthread_mutex_unlock(&lock_directory_handles_);
-    perf::Dec(file_system_->no_open_dirs());
```

Comment:
Here I am not sure, since the call to perf::Dec was not in the protected section... Move it outside the protected section was quite complex.

Reply:
That should be fine.